sdks/python/apache_beam/runners/runner.py

```diff
@@ -158,181 +177,39 @@ def run_pipeline(
     # type: (...) -> PipelineResult
 
     """Execute the entire pipeline or the sub-DAG reachable from a node.
-
-    Runners should override this method.
     """
-    raise NotImplementedError
+    pipeline.visit(
+        group_by_key_input_visitor(
+            not options.view_as(TypeOptions).allow_non_deterministic_key_coders)
+    )
+
+    # TODO: https://github.com/apache/beam/issues/19168
+    # portable runner specific default
+    if options.view_as(SetupOptions).sdk_location == 'default':
+      options.view_as(SetupOptions).sdk_location = 'container'
+
+    return self.run_full_pipeline(
```

Comment:
What is the semantic  distinction between `run_pipeline` vs `run_full_pipeline`? It sounds like `run_pipeline` could run exectute subgraphs, but it calls into `run_full_pipeline`, which is supposed to run the entire graph.

Reply:
Mostly it's just a type distinction, but for backwards compatibility and the fact that only names (not type signatures) are used to resolve methods in Python I needed to call it something different. (IIRC, the old version could execute subgraphs at some point, I don't know if anyone uses that capability anymore.)

Reply:
i see, thanks. the only alternative that comes to mind is `run_portable_pipeline()`, but not sure if that would be a better name.

Reply:
Sorry for crashing, but I got tripped up by this already when writing code on top of these changes. I think any of `run_portable_pipeline` / `run_pipeline_proto` / `run_pipeline_from_proto` would be a bit clearer

Reply:
cc: @robertwb